Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +433,40 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid, false, true);
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.RequireOxygen)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in an oxygenless environment, put the fire out.
+                        if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
                     }
 
                     var source = EnsureComp<IgnitionSourceComponent>(uid);
                     _ignitionSourceSystem.SetIgnited((uid, source));
 
                     if (TryComp(uid, out TemperatureComponent? temp))
-                        _temperatureSystem.ChangeHeat(uid, 12500 * flammable.FireStacks, false, temp);
+                    {
+                        // get hot
+                        var maxDeltaT = flammable.PeakFlameTemperature - temp.CurrentTemperature;
+                        if (maxDeltaT > 0) // we dont want to burn and get colder.
+                        {
+                            var tempBump = MathF.Sqrt(maxDeltaT) * flammable.FireStacks * 25; //monotonic function, slows as it reaches maxT
+                            _temperatureSystem.ChangeHeat(uid, tempBump, false, temp);
+                        } 
+
+                        // release gas
+                        if (air != null && flammable.EmissiveGasMix != null)
+                        {
+                            flammable.EmissiveGasMix.Temperature = temp.CurrentTemperature;
+                            var releasingGas = new GasMixture(flammable.EmissiveGasMix.Volume);
+                            _atmosphereSystem.Merge(releasingGas, flammable.EmissiveGasMix); // generate gas for the entity's entire burning lifespan, but we use ReleaseGasTo to handle temperature/pressure for us since those will change for other reasons.
```

Comment:
Why not use `EmissiveGasMix.Clone()`?

Reply:
That's probably better! I just didn't see it. Will do.